easygems/healpix/__init__.py

```diff
@@ -84,17 +85,34 @@ def healpix_resample(var, xlims, ylims, nx, ny, src_crs, method="nearest", nest=
     res = np.full(latlon.shape[:-1], np.nan, dtype=var.dtype)
 
     if method == "nearest":
-        pix = healpy.ang2pix(
-            healpy.npix2nside(len(var)),
+        pix = healpix.ang2pix(
+            healpix.npix2nside(len(var)),
             theta=points[0],
             phi=points[1],
             nest=nest,
             lonlat=True,
         )
         res[valid] = var[pix]
     elif method == "linear":
-        res[valid] = healpy.get_interp_val(
-            np.asanyarray(var), theta=points[0], phi=points[1], nest=True, lonlat=True
+        lons, lats = healpix.pix2ang(
+            nside=healpix.npix2nside(len(var)),
+            ipix=np.arange(len(var)),
```

Comment:
This line will compute angles for all the points, so the scaling will be proportional to the size of the input grid. I don't see an immediate fix using what `healpix` provides (we'd probably at least need a function computing the neighbor indices). But we should keep that in mind.

Reply:
So far the following `griddata` is by far the more severe bottleneck. But I agree that we should also keep this in mind, especially if we opt for ideas like storing high-resolution nested simulations on "cut out" HEALPix grids.